aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -217,6 +217,38 @@ Tensor sparse_csr_tensor(
       options.pinned_memory_opt());
 }
 
+Tensor empty_sparse_csr(
+    IntArrayRef size,
+    c10::optional<ScalarType> dtype,
+    c10::optional<Layout> layout,
+    c10::optional<Device> device,
+    c10::optional<bool> pin_memory,
+    c10::optional<MemoryFormat> optional_memory_format) {
+  check_size_nonnegative(size);
+
+  TORCH_CHECK(size.size() == 2, "torch.empty: Only 2D sparse CSR tensors are supported.");
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(layout == Layout::SparseCsr);
+
+  auto rows = size[0];
+  auto cols = size[1];
+  int64_t nnz = 0;
+
+  TensorOptions options = TensorOptions().dtype(ScalarType::Long).layout(Layout::Strided).device(device).pinned_memory(pin_memory);
+  auto crow_indices = at::empty({rows + 1}, options);
```

Comment:
If you call empty for a strided Tensor you still get valid stride and size, but here empty would cause crow_indices and col_indices to be caused with values that could not be within bounds. Should we instead use something like arange to generate very basic, but valid indices for this?

Reply:
Empty for strided Tensor has garbage as values. `crow_indices` must have the size `rows+1`, size of `crow_indices` changes only under resize operation, otherwise, it's fixed and its actual content is tied to values and column indices. I don't think it's a problem that `empty` returns crow_indices with garbage, it's used for allocating the memory and then filling it.

Reply:
Yes, the content of the memory for strided Tensor is garbage, but the size and stride fields are still valid. To some extend the indices here are extensions of these, since they're part of the storage as well. Even an empty strided Tensor won't run into segfaults if indexed for any sequence of valid indices. In this case however we might end up with elements in the row and col indices that cause segfaults when used to index into the values Tensor. So there exists an index for a sparse CSR tensor that causes a segfault, which does not exist for strided Tensors.